operator/internal/controller/redpanda/role_controller.go

```diff
@@ -86,13 +87,32 @@ func (r *RoleReconciler) SyncResource(ctx context.Context, request ResourceReque
 			return createPatch(err)
 		}
 		hasManagedRole = true
+		hasManagedPrincipals = shouldManagePrincipals
+	}
+
+	if hasRole && shouldManageRole {
+		// Update principals if we should manage them
+		if shouldManagePrincipals {
+			if err := rolesClient.Update(ctx, role); err != nil {
+				return createPatch(err)
+			}
+			hasManagedPrincipals = true
+		} else if hasManagedPrincipals {
+			// If we were managing principals but shouldn't anymore, clear them
+			if err := rolesClient.ClearPrincipals(ctx, role); err != nil {
+				return createPatch(err)
+			}
+			hasManagedPrincipals = false
+		}
+		hasManagedRole = true
 	}
 
 	if hasRole && !shouldManageRole {
 		if err := rolesClient.Delete(ctx, role); err != nil {
 			return createPatch(err)
 		}
 		hasManagedRole = false
+		hasManagedPrincipals = false
 	}
```

Comment:
As far as I can see this is dead code as `shouldManageRole` will always be set to `true`